### ISSUE TITLE
Should I expect AutoPublishCodeSha256 in Globals in next releases?

### ISSUE THREAD

**Issue by marcoblos** [awslabs/serverless-application-model#1550]
tldr;

- will `AutoPublishCodeSha256` become a `Globals` property as `AutoPublichAlias`?
- is there a feature request to `Lambda Version` become editable instead of a sequence like 1, 2, 3, ...N?

-------

Hi!

I'm creating SAM templates using `AutoPublishCodeSha256` and `AutoPublishAlias` features.

Right now, I know I can use `AutoPublishAlias` in `Globals`, but I would like to know if there is any possibility in the future to also use `AutoPublishCodeSha256` in `Globals`.

My point here, I would like to track my changes, matching them with a Pipeline ID, Git Tag or Git Hash generated by a commit. Right now to do that I need to repeat the `AutoPublishCodeSha256` in every `Type: AWS::Serverless::Function` what is quite boring, repetitive and lead to some errors.

I also understand that this `AutoPublishCodeSha256` was not intended to be a way to control the lambda version, but a way to "help the developers control when the lambda should be replaced".

If I'm misunderstanding something or you need some clarification , just say it.
🙏

**Comment by hawflau** [comment]
Hi @marcoblos, 

answering your questions:
> will AutoPublishCodeSha256 become a Globals property as AutoPublichAlias?

Can you share your sam template for us to understand more and assess potential impacts?  It should be an easy add, but we'd like understand more about the use case first.

> is there a feature request to Lambda Version become editable instead of a sequence like 1, 2, 3, ...N?
The version number seems to be auto-increasing sequence and is not editable (as documented in [Lambda's doc](https://docs.aws.amazon.com/lambda/latest/dg/configuration-versions.html). For `AWS::Serverless::Function` resource, you can use `AutoPublishAlias` to specify an alias.
